addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -3239,10 +3312,15 @@ const SnippetOptionWidget = Widget.extend({
 
             let obj = this;
             if (params.applyTo) {
-                const $firstSubTarget = this.$(params.applyTo).eq(0);
+                let $firstSubTarget = this.$(params.applyTo).eq(0);
                 if (!$firstSubTarget.length) {
-                    widget.toggleVisibility(false);
-                    return;
+                    if (params.applyToOrCreate) {
+                        this[params.applyToOrCreate]();
+                        $firstSubTarget = this.$(params.applyTo).eq(0);
+                    } else {
+                        widget.toggleVisibility(false);
+                        return;
+                    }
```

Comment:
Oh... I was actually joking when talking about this applyToOrCreate to be generic but you actually tried. Why not but in any case, look where you are doing it :fearful: `updateUIVisibility` is updating the editor panel to make its widget visible or not... and if it does not find an element, *it creates it and thus **modifies** the DOM*. Obviously this is wrong: those are different concepts but mailing: editing the DOM should only be done at the right time (during option method calls).

I think you will ease your pain a lot by not making this generic.